File: src/cli/systemctl.go
package cli

import (
	"fmt"
	"strings"
)

// SystemCtl wraps systemctl commands
type SystemCtl struct {
	serviceName string
}

// NewSystemCtl creates a new SystemCtl command builder
func NewSystemCtl() *SystemCtl {
	return &SystemCtl{}
}

// Service sets the service name
func (s *SystemCtl) Service(name string) *SystemCtl {
	s.serviceName = name
	return s
}

// Start generates systemctl start command
func (s *SystemCtl) Start() string {
	return fmt.Sprintf("systemctl start %s || true", s.serviceName)
}

// Stop generates systemctl stop command
func (s *SystemCtl) Stop() string {
	return fmt.Sprintf("systemctl stop %s || true", s.serviceName)
}

// Enable generates systemctl enable command
func (s *SystemCtl) Enable() string {
	return fmt.Sprintf("systemctl enable %s || true", s.serviceName)
}

// Disable generates systemctl disable command
func (s *SystemCtl) Disable() string {
	return fmt.Sprintf("systemctl disable %s || true", s.serviceName)
}

// Restart generates systemctl restart command
func (s *SystemCtl) Restart() string {
	return fmt.Sprintf("systemctl restart %s || true", s.serviceName)
}

// Status generates systemctl status command (non-interactive, no pager)
func (s *SystemCtl) Status() string {
	// Disable systemd pager to avoid hanging in interactive mode over SSH/PTTY
	// and force --no-pager so output returns immediately.
	return fmt.Sprintf("SYSTEMD_PAGER= systemctl status %s --no-pager", s.serviceName)
}

// IsActive generates systemctl is-active command
func (s *SystemCtl) IsActive() string {
	return fmt.Sprintf("systemctl is-active %s", s.serviceName)
}

// ParseIsActive parses output to check if service is active
func ParseIsActive(output string) bool {
	return strings.TrimSpace(output) == "active"
}

File: csharp-enva/CLI/SystemCtl.cs
namespace Enva.CLI;

public class SystemCtl
{
    private string? serviceName;

    public static SystemCtl NewSystemCtl()
    {
        return new SystemCtl();
    }

    public SystemCtl Service(string name)
    {
        serviceName = name;
        return this;
    }

    public string Restart()
    {
        if (string.IsNullOrEmpty(serviceName))
        {
            throw new Exception("Service name must be set");
        }
        return $"systemctl restart {serviceName}";
    }

    public string Enable()
    {
        if (string.IsNullOrEmpty(serviceName))
        {
            throw new Exception("Service name must be set");
        }
        return $"systemctl enable {serviceName}";
    }

    public string IsActive()
    {
        if (string.IsNullOrEmpty(serviceName))
        {
            throw new Exception("Service name must be set");
        }
        return $"systemctl is-active {serviceName}";
    }

    public string Start()
    {
        if (string.IsNullOrEmpty(serviceName))
        {
            throw new Exception("Service name must be set");
        }
        return $"systemctl start {serviceName}";
    }

    public string Status()
    {
        if (string.IsNullOrEmpty(serviceName))
        {
            throw new Exception("Service name must be set");
        }
        return $"systemctl status {serviceName}";
    }

    public string Stop()
    {
        if (string.IsNullOrEmpty(serviceName))
        {
            throw new Exception("Service name must be set");
        }
        return $"systemctl stop {serviceName}";
    }

    public string Disable()
    {
        if (string.IsNullOrEmpty(serviceName))
        {
            throw new Exception("Service name must be set");
        }
        return $"systemctl disable {serviceName}";
    }

    public static bool ParseIsActive(string output)
    {
        return output.Trim() == "active";
    }
}
